app.py

```diff
@@ -0,0 +1,75 @@
+from bson.json_util import dumps
+from bson.objectid import ObjectId
+from flask import Flask, render_template, request, redirect, url_for
+from pymongo import MongoClient
+
+import datetime
+import json
+import os
+import sys
+
+client = MongoClient(os.environ['MONGODB_URI'])
```

Comment:
It's arguably bad practice to do actual work at module level in Python; anytime somebody imports this module it will start doing things and talking to the network immediately, which can prevent importing from succeeding if anything goes wrong. The common tactic is instead to at least wrap in a function (named, say, `make_client` or `connect` or whatnot) then call that function where necessary, inside your tasks/endpoints.

That said - it sounds like MongoClient does all of its networky things in a background thread as of 3.0, and it also does connection pooling and such, so having a single instance setup and shared at module level is probably not that great a sin. But it does teach anybody new to Python bad habits :)

Reply:
Wouldn't it add quite a bit of overhead to connect and disconnect each time an endpoint is called?

Reply:
That is what you mean for me to do - right?